### ISSUE TITLE
Test standalone `git clone`-based installation in CI

### ISSUE THREAD

**Issue by mguaypaq** [spinalcordtoolbox/spinalcordtoolbox#4054]
This was originally mentioned in the context of https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4049#discussion_r1117539482.

To summarize:

- Both `install_sct` and `install_sct.bat` can be run on their own, without any other source files present. 
  - Doing so will `git clone` a specific version of SCT to a tempdir. The source files will then be copied to the desired installation directory, once it is chosen by the user.
- For the case of installing SCT in CI runs on PRs:
  - Both scripts have CLI options to pass a specific git ref. 
  - So for our CI, in theory, we could copy the install script to a dedicated directory, then pass `${{github.event.pull_request.head.sha}}`, and the script would `git clone` the PR's branch to a tempdir and install from there.

However, we've tried this before, and faced a big blocker:

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3779
- tl;dr: `${{ github.event.pull_request.head.sha }}` resolves to a SHA that's present in the fork, but not present in the main repository. So, when the installer scripts try to `git clone` this SHA from the main SCT repository, the workflow fails.

The way we work around this is by letting `actions/checkout` properly do the git clone from the fork, then do an [in-place install](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3797) instead. But, in doing so, we fail to test the script's own "`git clone` to a tempdir" path.

**Comment by joshuacwnewton** [comment]
@mguaypaq [suggested](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4049#discussion_r1119278543) that we don't bother trying to make this work for PRs, and instead add `git clone`-based installation testing to the `master` branch (e.g. on merge, on cron jobs, etc.)

- This may let some subtle bugs slip through in PRs, but it would at least alert us to said bugs once the PR is merged, which I think is a good enough line of defense.
- Using `master` would also skip the requirement for a "git ref" CLI option, meaning we could remove that logic from both scripts.

**Comment by joshuacwnewton** [comment]
Reopening this issue as the merged PR does not work expected for:

- Windows (Native)
   - The check `if %GITHUB_REF_NAME%=="master" (` doesn't seem to properly fire on `master`. (wrong bat syntax? `GITHUB_REF_NAME` different on windows?)
- WSL
   - The `.ci.sh` script fails with `./.ci.sh: line 17: GITHUB_REF_NAME: unbound variable` (is `GITHUB_REF_NAME` not present on `wsl-bash`?)

**Comment by joshuacwnewton** [comment]
Quick investigation: 

- For Windows, it looks like all that might be needed is [double-quoting the environment variable](https://stackoverflow.com/a/14955571).
- For WSL, I don't really have a strong understanding of how `Vampire/setup-wsl` works, so I don't really get why the internal Ubuntu VM doesn't have access to GitHub's default environment variables. :thinking:

**Comment by joshuacwnewton** [comment]
Ah, wait, here we are: https://github.com/timescale/timescaledb/actions/runs/5004516172/workflow#L263-L274

```yaml
        # GitHub Actions allow you neither to use the env context for the job name,
        # nor to access the job name from the step context, so we have to
        # duplicate it to work around this nonsense.
        # [...]
        GITHUB_REF_NAME: ${{ github.ref_name }}
        # [...]
```

I'm going to give these fixes a shot.

**Comment by joshuacwnewton** [comment]
1/2. Windows worked, WSL didn't. Going to try one more fix.

**Comment by joshuacwnewton** [comment]
Aha! It worked. 

((I made the mistake of only updating "WSL [Ubuntu 20.04]", though. I'll go ahead and update both, and this issue should be good to close.))
